Source/AppAuthTV/OIDTVTokenRequest.h

```diff
@@ -0,0 +1,79 @@
+   @copydetails
+       Licensed under the Apache License, Version 2.0 (the "License");
+       you may not use this file except in compliance with the License.
+       You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+       Unless required by applicable law or agreed to in writing, software
+       distributed under the License is distributed on an "AS IS" BASIS,
+       WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+       See the License for the specific language governing permissions and
+       limitations under the License.
+*/
+
+#import "OIDTokenRequest.h"
+
+#import <Foundation/Foundation.h>
+
+@class OIDServiceConfiguration;
+@class OIDTVServiceConfiguration;
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface OIDTVTokenRequest : OIDTokenRequest
+
+/*! @brief The device code received from the authorization server.
+    @remarks device_code
+    @see https://tools.ietf.org/html/rfc8628#section-3.4
+ */
+@property(nonatomic, readonly, nullable) NSString *deviceCode;
+
+- (instancetype)init NS_UNAVAILABLE;
+
+- (instancetype)initWithConfiguration:(OIDServiceConfiguration *)configuration
+                            grantType:(NSString *)grantType
+                    authorizationCode:(nullable NSString *)code
+                          redirectURL:(nullable NSURL *)redirectURL
+                             clientID:(NSString *)clientID
+                         clientSecret:(nullable NSString *)clientSecret
+                               scopes:(nullable NSArray<NSString *> *)scopes
+                         refreshToken:(nullable NSString *)refreshToken
+                         codeVerifier:(nullable NSString *)codeVerifier
+                 additionalParameters:
+                     (nullable NSDictionary<NSString *, NSString *> *)additionalParameters
+    NS_UNAVAILABLE;
+
+- (instancetype)initWithConfiguration:(OIDServiceConfiguration *)configuration
+                            grantType:(NSString *)grantType
+                    authorizationCode:(nullable NSString *)code
+                          redirectURL:(nullable NSURL *)redirectURL
+                             clientID:(NSString *)clientID
+                         clientSecret:(nullable NSString *)clientSecret
+                                scope:(nullable NSString *)scope
+                         refreshToken:(nullable NSString *)refreshToken
+                         codeVerifier:(nullable NSString *)codeVerifier
+                 additionalParameters:
+                     (nullable NSDictionary<NSString *, NSString *> *)additionalParameters
+    NS_UNAVAILABLE;
+
+- (instancetype)initWithConfiguration:(OIDTVServiceConfiguration *)configuration
+                            grantType:(NSString *)grantType
+                           deviceCode:(NSString *)deviceCode
+                             clientID:(NSString *)clientID
+                         clientSecret:(NSString *)clientSecret
```

Comment:
Is clientSecret needed for Device Authorization access token request?

Reply:
It is not. In the parent class, if clientsecret is present it will do HTTP Basic Authentication otherwise it will just send the client ID in the request body. https://github.com/openid/AppAuth-iOS/blob/master/Source/AppAuthCore/OIDTokenRequest.m#L277